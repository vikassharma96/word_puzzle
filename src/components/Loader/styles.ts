import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';

const styles = StyleSheet.create({
  overlay: {
    zIndex: 1,
    height: '100%',
    width: '100%',
    opacity: 0.8,
    alignItems: 'center',
    position: 'absolute',
    justifyContent: 'center',
    backgroundColor: colors.default.background,
  },
});

export default styles;
