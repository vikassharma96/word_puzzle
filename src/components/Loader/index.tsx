import React from 'react';
import {View, ActivityIndicator} from 'react-native';
import styles from './styles';

const Loader = ({visible = false}) => {
  return (
    <>
      {visible && (
        <View style={styles.overlay}>
          <ActivityIndicator size={'large'} />
        </View>
      )}
    </>
  );
};

export default Loader;
