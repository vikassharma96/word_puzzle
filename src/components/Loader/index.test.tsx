import {render} from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '.';

describe('@components/Loader', () => {
  const component = <Loader visible={true} />;

  it('should match snapshots', () => {
    const componentTree = renderer.create(component).toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  it('should be able to hide the loader', () => {
    const element = render(<Loader />).toJSON();
    expect(element).toBeNull();
  });
});
