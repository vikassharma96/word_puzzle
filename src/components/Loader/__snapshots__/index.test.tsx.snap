// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Loader should match snapshots 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "height": "100%",
      "justifyContent": "center",
      "opacity": 0.8,
      "position": "absolute",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;
